src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -244,6 +244,17 @@ struct CylindricalPoint{T} <: AbstractCylindricalPoint{T}
     CylindricalPoint{T}(r::Real, φ::Real, z::Real) where {T} = new(T(r), mod(T(φ),T(2π)), T(z))
 end
 
+# Handles Unitful quantities or plain numbers 
+function CylindricalPoint(r, φ, z)
+    r_val = isa(r, Unitful.Quantity) ? float(ustrip(uconvert(u"m", r))) : float(r)
+    φ_val = isa(φ, Unitful.Quantity) ? float(ustrip(uconvert(u"rad", φ))) : float(φ)
+    z_val = isa(z, Unitful.Quantity) ? float(ustrip(uconvert(u"m", z))) : float(z)
```

Comment:
`ustrip(unit, x)` is the same as `ustrip(uconvert(unit,x))` and also use internal units where intended:
```suggestion
    r_val = isa(r, Unitful.Quantity) ? float(ustrip(internal_length_unit, r))) : float(r)
    φ_val = isa(φ, Unitful.Quantity) ? float(ustrip(internal_angle_unit, φ))) : float(φ)
    z_val = isa(z, Unitful.Quantity) ? float(ustrip(internal_length_unit, z))) : float(z)
```

Reply:
Also, is `float` what we want?
If one of them is an `Int`, `float` will make this a `Float64`

I would argue, though, that if you pass `Int, Float32, Float32`, you would like your final point to be of type `Float32` rather than `Float64`.